bitcoin/src/lib.rs

```diff
@@ -13,21 +13,25 @@
 //! See README.md for detailed documentation about development and supported
 //! environments.
 //!
-//! ## Available feature flags
+//! # Features
 //!
-//! * `std` - the usual dependency on `std` (default).
-//! * `secp-recovery` - enables calculating public key from a signature and message.
-//! * `base64` - (dependency), enables encoding of PSBTs and message signatures.
-//! * `rand` - (dependency), makes it more convenient to generate random values.
-//! * `serde` - (dependency), implements `serde`-based serialization and
-//!                 deserialization.
+//! * `default` - enables `std` and `secp-recovery`.
+//! * `std` - the usual dependency on `std`.
+//! * `serde` - implements `serde`-based serialization and deserialization.
 //! * `secp-lowmemory` - optimizations for low-memory devices.
-//! * `bitcoinconsensus-std` - enables `std` in `bitcoinconsensus` and communicates it
-//!                            to this crate so it knows how to implement
-//!                            `std::error::Error`. At this time there's a hack to
-//!                            achieve the same without this feature but it could
-//!                            happen the implementations diverge one day.
-//! * `ordered` - (dependency), adds implementations of `ArbitraryOrdOrd` to some structs.
+//! * `secp-recovery` - enables calculating public key from a signature and message.
```

Comment:
IIRC this is required for signmessage, so the doc string should be "enables signmessage". But maybe we should rename the feature too. I'm also tempted to put it into a separate crate since it's not related to most of the Bitcoin stuff.

Reply:
All valid points, punting to another PR so I don't have to dig into it right now.